File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        i=len(nums)-2
        dp = [0]*i
        dest = len(nums)-1
        while i>=0:
            if dest-i <= nums[i]:
                dest = i
            i=i-1
        
        if dest == 0:
            return True
        return False
        

        
        

File: 55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
       int steps = 0;
       for(int i=nums.size()-2; i>=0; i--){
            steps++;
            if(steps<=nums[i]){
                steps = 0;
            }
       }     
       return steps==0;
    }
};